src/ReflectionParameter.php

```diff
@@ -365,6 +374,23 @@ public function isVariadic()
     }
 
     /**
+     * Returns if default value set (or implied).
+     *
+     * Identical to isDefaultValueAvailable(), except it
+     * includes IMPLIED default values which:
+     *     + Only exist in builtin functions and methods.
+     *     + Only affect the optionality of a prameter, not
+     *         if a SUPPLIED parameter can have the default
+     *         value.
+     *
+     * @return bool
+     */
+    protected function isDefaultValueSet()
```

Comment:
1. Any cases, when `isDefaultValueSet` would work better, than calling `isDefaultValueAvailable`?
2. Maybe instead we should improve `isDefaultValueAvailable`?

Reply:
Again, this is part of the handling for optional parameters without default values, which is new in this PR. Perhaps I should make `isDefaultValueSet()` private so it's more clearly a internal convienience function, rather than something that's part of the interface.

Reply:
Better to move this into separate PR for optional parameter without default value handling (part 2).

Reply:
OK. See above.